### DISCUSSION TITLE
How Internal state machine works in react query?

### DISCUSSION THREAD

**Discussion by akm021** [TanStack/query#6120] [Q&A]
Hi all, I am trying to figure out how React Query actually works in a component. (I think it's more like a React-related doubt.)

React Query exposes a bunch of boolean flags, which are derived from the internal state machine. While consuming these boolean flags, will the components create a local copy of those states? How does the re-render logic actually work when a flag gets changed? Is there any setState or props change kind of mechanism involved at the component level?

**Comment by TkDodo** [comment]
> While consuming these boolean flags, will the components create a local copy of those states

no

> How does the re-render logic actually work when a flag gets changed

The state lives in a store outside of react. We then inform the components about updates via the `useSyncExternalStore` hook from react.

**Comment by akm021** [comment]
Thanks for the clarification @TkDodo
